langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStore.java

```diff
@@ -27,36 +33,74 @@
  * Only cosine similarity is used.
  * Only ivfflat index is used.
  */
+@NoArgsConstructor(force = true) // Needed for inherited bean injection validation
 public class PgVectorEmbeddingStore implements EmbeddingStore<TextSegment> {
-
     private static final Logger log = LoggerFactory.getLogger(PgVectorEmbeddingStore.class);
+    /**
+     * Datasource used to create the store
+     */
+    protected final DataSource datasource;
+    /**
+     * Embeddings table name
+     */
+    protected final String table;
+    /**
+     * Metadata handler
+     */
+    protected final MetadataHandler metadataHandler;
 
-    private static final Gson GSON = new Gson();
+    /**
+     * Constructor for PgVectorEmbeddingStore Class
+     *
+     * @param datasource            The datasource to use
+     * @param table                 The database table
+     * @param dimension             The vector dimension
+     * @param useIndex              Should use <a href="https://github.com/pgvector/pgvector#ivfflat">IVFFlat</a> index
+     * @param indexListSize         The IVFFlat number of lists
+     * @param createTable           Should create table automatically
+     * @param dropTableFirst        Should drop table first, usually for testing
+     * @param metadataConfig        The {@link MetadataConfig} config.
+     */
+    @Builder
+    protected PgVectorEmbeddingStore(DataSource datasource,
+                                  String table,
+                                  Integer dimension,
+                                  Boolean useIndex,
+                                  Integer indexListSize,
+                                  Boolean createTable,
+                                  Boolean dropTableFirst,
+                                  MetadataConfig metadataConfig) {
+        this.datasource = datasource;
+        this.table = ensureNotBlank(table, "table");
+        MetadataConfig config = Optional.ofNullable(metadataConfig).orElse(DefaultMetadataConfig.defaultConfig());
```

Comment:
```suggestion
        MetadataConfig config = getOrDefault(metadataConfig, DefaultMetadataConfig::defaultConfig);
```
minor, to keep the same style